Enigmatry.Entry.AspNetCore.Tests/HttpContextBuilder.cs

```diff
@@ -59,12 +63,29 @@ internal HttpContext Build()
 
         var provider = new ServiceCollection()
             .AddTransient(_ => A.Fake<IHostEnvironment>())
-            .AddTransient(_ => A.Fake<ILogger<ExceptionHandler>>())
+            .AddTransient<ILogger<ExceptionHandler>>(_ => new NullLogger<ExceptionHandler>())
+            .AddTransient<IActionResultExecutor<JsonResult>, SystemTextJsonResultActionExecutor>()
             .BuildServiceProvider();
 
         return new DefaultHttpContext(features)
         {
             RequestServices = provider
         };
     }
+
+    internal class SystemTextJsonResultActionExecutor : IActionResultExecutor<JsonResult>
```

Comment:
@typhoon41 After this pull request completes, please revisit all the tests that were created for the purpose of setting up HttpContext. All the setup and mocking that was needed in the HttpContextBuilder feels like we are reinventing internal startup code from the asp net pipeline. Why do we need to know about FeatureCollection, StreamResponseBodyFeature, HostEnvironment, IHttpRequestFeature,  SystemTextJsonActionResultExecutor, streaming? These tests do not guarantee that this code will work in reall app since it still needs to be integrated into the ASP pipeline and if we write integration test for it (as we have now), then the cost of maintaining this code is no longer justified.

Reply:
Well now...If we wanted to cover those classes with unit tests (which should be the easiest task comparing with other types of tests at least on paper), this had to be done. I agree it's better to test this through integration/system tests, but these tests were ported from EBO project. There (until recently) we only had unit tests, so this wasn't written from scratch.

The more tests we have, the bigger our confidence is. I agree we shall be ditching them if it appears we need to sweat a bit to maintain them, but right now they're working without too much investment from our side.